test/scripts/e2e_go_tests.sh

```diff
@@ -91,7 +91,14 @@ fi
 
 echo "PARALLEL_FLAG = ${PARALLEL_FLAG}"
 
-if [ "${#TESTPATTERNS[@]}" -eq 0 ]; then
+if [ "$USE_CIRCLECI_SPLIT" != "" ]; then
```

Comment:
Why use circleci tests split instead of the partitioning?

Reply:
I'm not actually sure what this change is, could one of you two explain? @algonautshant @cce

Reply:
oh, this is from when I was evaluating the "circleci tests split" runner before Shant's partitioning stuff was ready. you can remove these lines to get rid of it .. as you noticed the `elif [ "${#TESTPATTERNS[@]}" -eq 0 ]; then` is the regular code path we have used and the partitioning stuff is now using.

Reply:
You would still need to keep the `gotestsum` call, or set `GOTESTCOMMAND` to include the junit stuff

Reply:
the entire `if USE_CIRCLECI_SPLIT` section can be removed. This file doesn't need to be changed for this PR.

Reply:
@bricerisingalgorand could you please apply this to your branch? https://github.com/bricerisingalgorand/go-algorand/commit/5a1b71eec5b229ba906442bafce8fcc5cbc3e330